app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivityViewModel.kt

```diff
@@ -51,7 +52,7 @@ class DeviceShieldTrackerActivityViewModel @Inject constructor(
     private val vpnFeatureRemover: VpnFeatureRemover,
     private val vpnStore: VpnStore,
     private val dispatcherProvider: DispatcherProvider,
-) : ViewModel() {
+) : ViewModel(), NotifyMeListener {
```

Comment:
The component can be displayed in different screens. If you dismiss from one screen it should still be visible in the others. To implement this logic inside the view it needs to know the context where it is displayed.

I see only one way to do it. 
Add a new `screenName` custom attribute which must be unique and equal to the screen name. Internally the component will use its own shared preferences to track the dismiss state for each key. It will also send the pixel with the `screenName` provided (as per the latest requirement).

The last function in the interface `visibilityChanged` can be replaced by a `onVisibilityChanged(listener)` function defined in the component. This is similar to how Android provides other listeners.

Is this what you had in mind, or something else?